projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/execution/JobExecutionUtil.java

```diff
@@ -0,0 +1,113 @@
+import org.apache.commons.lang3.StringUtils;
+
+import com.vmware.taurus.service.model.ExecutionStatus;
+import com.vmware.taurus.service.model.ExecutionTerminationMessage;
+import com.vmware.taurus.service.model.ExecutionTerminationStatus;
+
+@Slf4j
+public class JobExecutionUtil {
+
+   static final String TERMINATION_MESSAGE_ATTRIBUTE_STATUS = "status";
+   static final String TERMINATION_MESSAGE_ATTRIBUTE_VDK_VERSION = "vdk_version";
+
+   public static ExecutionTerminationMessage getTerminationMessage(ExecutionStatus executionStatus, String terminationMessage) {
+      String vdkVersion = "";
+      String terminationStatus = "";
+
+      if (!StringUtils.isEmpty(terminationMessage)) {
+         try {
+            Map<String, String> obj = new Gson().fromJson(terminationMessage, Map.class);
+            terminationStatus = obj.get(TERMINATION_MESSAGE_ATTRIBUTE_STATUS);
+            vdkVersion = obj.getOrDefault(TERMINATION_MESSAGE_ATTRIBUTE_VDK_VERSION, "");
+         } catch (com.google.gson.JsonSyntaxException ex) {
+            // Fallback to the old plain text format
+            terminationStatus = terminationMessage;
+         }
+      }
+
+      return ExecutionTerminationMessage.builder()
+            .terminationStatus(getTerminationStatus(executionStatus, terminationStatus))
+            .vdkVersion(vdkVersion)
+            .build();
+   }
+
+   public static ExecutionStatus getExecutionStatus(Boolean executionSucceeded) {
+      ExecutionStatus executionStatus;
+
+      if (executionSucceeded == null) {
+         executionStatus = ExecutionStatus.RUNNING;
+      } else if (executionSucceeded) {
+         executionStatus = ExecutionStatus.FINISHED;
+      } else {
+         executionStatus = ExecutionStatus.FAILED;
+      }
+
+      return executionStatus;
+   }
+
+   static ExecutionTerminationStatus getTerminationStatus(ExecutionStatus executionStatus, String terminationMessage) {
+      ExecutionTerminationStatus terminationStatus = ExecutionTerminationStatus.NONE;
+
+      // if there is no termination message due to the missing K8S Pod
+      // the method sets an appropriate termination status based on the execution status
+      if (StringUtils.isEmpty(terminationMessage) && ExecutionStatus.FINISHED.equals(executionStatus)) {
+         terminationStatus = ExecutionTerminationStatus.SUCCESS;
+      } else if (StringUtils.isEmpty(terminationMessage) && ExecutionStatus.FAILED.equals(executionStatus)) {
+         terminationStatus = ExecutionTerminationStatus.PLATFORM_ERROR;
+      } else {
+         terminationStatus = Arrays.stream(ExecutionTerminationStatus.values())
+               .filter(status -> status.getString().equals(terminationMessage))
+               .findAny()
+               .orElse(terminationStatus);
+      }
+
+      return terminationStatus;
+   }
+
+   static ExecutionStatus updateExecutionStatusBasedOnTerminationStatus(ExecutionStatus executionStatus, ExecutionTerminationStatus terminationStatus) {
```

Comment:
Because of the dependencies:

1. The determination of TerminationStatus depends on the initial ExecutionStatus(Running, Finished, Failed)
2. Full determination of ExecutionStatus depends on TerminationStatus(SKIPPED, USER_ERROR)